kahuna/public/js/edits/service.js

```diff
@@ -328,10 +328,14 @@ service.factory('editsService',
 
 
     function batchUpdateMetadataField(images, field, value, editOption = overwrite.key) {
-        return trackAll($q, $rootScope, field, images, image => {
+        return trackAll($q, $rootScope, field, images, async (image) => {
             const newFieldValue = getNewFieldValue(image, field, value, editOption);
-            return updateMetadataField(image, field, newFieldValue, true);
-        },'images-updated');
+            const updated = await updateMetadataField(image, field, newFieldValue, true);
+            if (!updated) { // updateMetadataField returns false if no change
+                return image;
+            }
+            return updated;
```

Comment:
or possibly even (function wouldn't need to be async then)
```suggestion
            return updateMetadataField(image, field, newFieldValue, true).then(
            	updated => updated || image; // updateMetadataField returns false if no change
            );
```